tests/functional/z_cluster/nodes/test_non_ocs_taint_and_toleration.py

```diff
@@ -197,28 +223,55 @@ def test_non_ocs_taint_and_tolerations(self):
 
             configmap_obj.patch(params=params, format_type="merge")
             logger.info(f"Successfully added toleration to {configmap_obj.kind}")
+            if config.ENV_DATA["mcg_only_deployment"]:
+                logger.info("Wait some time after adding toleration for pods respin")
+                waiting_time = 60
+                logger.info(f"Waiting {waiting_time} seconds...")
+                time.sleep(waiting_time)
+                logger.info("Force delete all pods")
+                pod_list = get_all_pods(
+                    namespace=config.ENV_DATA["cluster_namespace"],
+                    exclude_selector=True,
+                )
+                for pod in pod_list:
+                    pod.delete(wait=False)
 
-        if config.ENV_DATA["mcg_only_deployment"]:
-            logger.info("Wait some time after adding toleration for pods respin")
-            waiting_time = 60
-            logger.info(f"Waiting {waiting_time} seconds...")
-            time.sleep(waiting_time)
-            logger.info("Force delete all pods")
-            pod_list = get_all_pods(
-                namespace=config.ENV_DATA["cluster_namespace"],
-                exclude_selector=True,
-            )
-            for pod in pod_list:
-                pod.delete(wait=False)
+    def test_non_ocs_taint_and_tolerations(self, nodes):
+        """
+        Test runs the following steps
+        1. Taint odf nodes with non-ocs taint
+        2. Set tolerations on storagecluster, subscription, configmap and ocsinit
+        3. check tolerations on all subscription yaml.
+        4. Check toleration on all odf pods.
+        5. Add Capacity.
+        6. Reboot one of the nodes and check toleration on all odf pods on that node.
+        7. Replace one of the nodes and check all odf pods on that node are running.
 
-        logger.info("After edit wait for some time for pods to respin as expected")
-        time.sleep(180)
-        assert wait_for_pods_to_be_running(timeout=900, sleep=15)
+        """
+
+        number_of_pods_before = len(
+            get_all_pods(namespace=config.ENV_DATA["cluster_namespace"])
+        )
+
+        logger.info("Apply custom taints and tolerations.")
+        self.apply_custom_taint_and_toleration()
+
+        retry((CommandFailed, TolerationNotFoundException), tries=5, delay=10,)(
+            check_toleration_on_subscriptions
+        )(toleration_key="xyz")
```

Comment:
any reason for specifically checking tolerations on subscriptions? check_toleration_on_pods is not enough?